packages/providers/upload-aws-s3/lib/index.js

```diff
@@ -10,19 +10,25 @@ const _ = require('lodash');
 const AWS = require('aws-sdk');
 
 module.exports = {
-  init(config) {
+  init({baseUrl, rootPath = "", s3Options,  ...legacyS3Options}) {
+    if(legacyS3Options && process.env.NODE_ENV !== 'production')
+      console.log("S3 configuration options passed at root level of the plugin's providerOptions is deprecated and will be removed in a future release. You wrap them inside the 's3Options:{}' property.")
```

Comment:
I recommend using the same system as node for deprecation. e.g https://www.contentful.com/blog/2016/11/02/node-v7-urls-deprecations-warnings-and-developer-experience/

Reply:
I would emit the warning even in production I think for consistency 🤔

Reply:
ping @pwizla & @derrickmehaffy that can help you with drafting a doc update for this :)

Reply:
should be fairly easy but provider documentation is in the package, not the docs repo (as much as I would rather see it in the docs repo). There shouldn't be much to do in the docs themselves.

Reply:
@alexandrebodin do you mean using the `process.emitWarning` for showing the deprecated message? I am not sure I get what you mean by `system`.

Reply:
hi yes this is what I meant you process.emitWarning :)